apps/src/codebridge/FileBrowser/prompts/openNewFolderPrompt.tsx

```diff
@@ -21,6 +21,32 @@ type OpenNewFilePromptArgsType = {
   sendCodebridgeAnalyticsEvent: (eventName: string) => unknown;
 };
 
+type ValidateInputArgs = {
+  folderName: string;
+  parentId: FolderId;
+  projectFolders: ProjectType['folders'];
+};
+
+export const validateNewFolderName = ({
+  folderName,
+  parentId,
+  projectFolders,
+}: ValidateInputArgs) => {
+  if (!folderName.length) {
+    return;
+  }
+  if (!validateFolderName(folderName)) {
+    return codebridgeI18n.invalidNameError();
+  }
+
+  const existingFolder = Object.values(projectFolders).some(
+    f => f.name === folderName && f.parentId === parentId
+  );
+  if (existingFolder) {
+    return codebridgeI18n.folderExistsError();
+  }
+};
+
```

Comment:
this was the inlined `validateInput` anonymous function passed into `showDialog`. Broken out to be easier to test, and will probably move to a separate location soon.